app/main.py

```diff
@@ -1,8 +1,32 @@
 class Person:
-    # write your code here
-    pass
+    people = {}  # Атрибут класса для хранения экземпляров по их имени
 
+    def __init__(self, name: str, age: int):
+        self.name = name
+        self.age = age
+        Person.people[name] = self  # Добавление экземпляра в словарь people
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_data: list) -> list:
+
+    person_list = []
+
+    # Первый проход: Создание экземпляров Person
+    for person_dict in people_data:
+        name = person_dict["name"]
```

Comment:
Ensure that the dictionary `person_dict` contains the key 'name' before accessing it to avoid potential KeyErrors.